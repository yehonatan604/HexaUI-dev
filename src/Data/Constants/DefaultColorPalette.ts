import { TColor } from "../Types/TColor";

const defaultColorPalette: TColor[] = [
    // variants
    { name: "standard", dark: "zinc-300", light: "zinc-800" },
    { name: "failure", dark: "red-300", light: "red-800" },
    { name: "success", dark: "green-300", light: "green-800" },
    { name: "warning", dark: "yellow-300", light: "yellow-800" },
    { name: "info", dark: "blue-300", light: "blue-800" },
    { name: "lime", dark: "lime-300", light: "lime-800" },
    { name: "cyan", dark: "cyan-300", light: "cyan-800" },
    { name: "purple", dark: "purple-300", light: "purple-800" },
    { name: "pink", dark: "pink-300", light: "pink-800" },

    // component specific
    { name: "quote", dark: "zinc-300", light: "zinc-900" },
    { name: "cite", dark: "zinc-400", light: "zinc-500" },
    { name: "citeHover", dark: "blue-300", light: "blue-600" },
    { name: "hr", dark: "zinc-500", light: "zinc-500" },
    { name: "navbarText", dark: "zinc-50", light: "zinc-50" },
    { name: "navbarBackground", dark: "zinc-800", light: "zinc-800" },
    { name: "navbarLink", dark: "zinc-200", light: "zinc-200", active: "amber-200" },
    { name: "textInput", dark: "zinc-200", light: "zinc-900", failure: "red-500", success: "green-500", ring: "sky-500", standard: "gray-400" },
    { name: "helperText", dark: "zinc-400", light: "zinc-500" },
];

export default defaultColorPalette;