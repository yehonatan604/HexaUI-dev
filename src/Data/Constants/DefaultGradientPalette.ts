import { TColorPalette } from "../Types/TColorPalette";

const defaultGradientPalette: TColorPalette = {
    primary: "from-blue-400 via-blue-600 via-blue-800 via-green-500 via-green-800 to-cyan-400",
    secondary: "from-purple-400 via-purple-600 via-purple-800 via-red-500 via-red-800 to-purple-800",
    standrad: "from-stone-800 via-stone-600 via-stone-500 via-netural-300 via-netural-200 to-zinc-500",
    info: "from-zinc-400 via-zinc-600 via-zinc-800 via-stone-500 via-yellow-stone to-slate-600",
    select: "from-lime-400 via-lime-600 via-lime-800 via-green-500 via-green-800 to-green-400",
    cancel: "from-rose-400 via-rose-600 via-rose-800 via-red-500 via-red-800 to-lime-600",
    approve: "from-indigo-400 via-indigo-600 via-indigo-800 via-yellow-500 via-yellow-800 to-blue-400",
    success: "from-green-400 via-green-600 via-green-800 via-violet-500 via-violet-800 to-lime-400",
    warning: "from-amber-400 via-amber-600 via-amber-800 via-orange-500 via-yellow-800 to-rose-400",
    failure: "from-red-400 via-red-600 via-zinc-200 via-zinc-100 to-red-700",
}

export default defaultGradientPalette;