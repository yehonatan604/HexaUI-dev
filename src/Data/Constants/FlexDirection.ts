export const FlexDir = {
    Row: "row",
    RowReverse: "row-reverse",
    Col: "col",
    ColReverse: "col-reverse"
}