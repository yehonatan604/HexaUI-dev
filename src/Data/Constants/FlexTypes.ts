export const FlexTypes = {
    Start: "start",
    End: "end",
    Center: "center",
    SpaceBetween: "between",
    SpaceAround: "around",
    SpaceEvenly: "evenly",
    Baseline: "baseline",
    Stretch: "stretch",
}

declare global {
    type FlexTypes = typeof FlexTypes[keyof typeof FlexTypes];
};