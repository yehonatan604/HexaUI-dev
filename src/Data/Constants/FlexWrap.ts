export const FlexWrap = {
    NoWrap: "nowrap",
    Wrap: "wrap",
    WrapReverse: "wrap-reverse"
}