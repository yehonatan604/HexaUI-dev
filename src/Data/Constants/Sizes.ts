export const Sizes = {
    "Xxs": "xxs",
    "Xs": "xs",
    "Sm": "sm",
    "Md": "md",
    "Lg": "lg",
    "Xl": "xl",
    "2Xl": "2xl",
    "3Xl": "3xl",
    "4Xl": "4xl",
}

export type Tsize = typeof Sizes;