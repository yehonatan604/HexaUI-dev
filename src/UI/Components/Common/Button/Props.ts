import { TComponent } from "../../../../Data/Types/TComponent";

export type ButtonProps = TComponent<"button"> & {
    variant?: string;
    size?: string;
    rounded?: boolean;
    roundedS?: boolean;
    roundedE?: boolean;
};