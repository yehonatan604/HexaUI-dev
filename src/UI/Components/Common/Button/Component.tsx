import useTheme from "../../../../Core/Hooks/useTheme";
import { Sizes } from "../../../../Data/Constants/Sizes";
import { TButton } from "../../../../Data/Types/ComponentTypes/Common/Button/TButton";

const Button = (props: TButton) => {
  const { options, children, className, ...componentProps } = props;
  const { colors } = useTheme();
  const size = options?.size;
  const rounded = options?.rounded || false;
  const roundedS = options?.roundedS || false;
  const roundedE = options?.roundedE || false;
  const bgColor = options?.bgVariant || colors.secondaryDark;
  const textColor = options?.textVariant || colors.select;

  let buttonSize: string;
  if (size === Sizes.Sm) {
    buttonSize = "px-2 py-1 text-sm";
  } else if (size === Sizes.Lg) {
    buttonSize = "px-4 py-2 text-lg";
  } else if (size === Sizes.Xl) {
    buttonSize = "px-5 py-3 text-xl";
  } else buttonSize = "px-4 py-3 text-base";

  let buttonRounded = "rounded";
  if (rounded) {
    buttonRounded = "";
  } else if (roundedS) {
    buttonRounded = "rounded-tl rounded-bl rounded-tr-0 rounded-br-0";
  } else if (roundedE) {
    buttonRounded = "rounded-tr rounded-br rounded-tl-0 rounded-bl-0";
  }

  return (
    <button
      className={`m-[1px] pb-[6px] py-[2.75px] ${buttonRounded} ${
        buttonSize || size
      } bg-${bgColor} text-${textColor} hover:bg-opacity-80 focus:outline-none focus:ring-2 focus:ring-opacity-50 focus:ring-gray-500 ${className}`}
      {...componentProps}
    >
      {children}
    </button>
  );
};

export default Button;
