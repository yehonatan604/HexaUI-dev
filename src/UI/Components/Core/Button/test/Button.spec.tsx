import { fireEvent, render, screen } from "@testing-library/react";
import { expect, test, vi } from "vitest";
import Button from "../Button";
import { Sizes } from "../../../../../Data/Constants/Sizes";

test("renders the button correctly", () => {
  render(<Button>Click Me</Button>);
  expect(screen.getByText("Click Me")).toBeInTheDocument();
});

test("matches snapshot", () => {
  const { asFragment } = render(<Button>Click Me</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("applies background color variant", () => {
  const { container } = render(<Button options={{ bgVariant: "success" }} />);
  expect(container.firstChild).toHaveClass("bg-success");
});

test("applies text color variant", () => {
  const { container } = render(<Button options={{ textVariant: "success" }} />);
  expect(container.firstChild).toHaveClass("text-success");
});

test("applies rounded styles when rounded prop is true", () => {
  const { container } = render(<Button options={{ notRounded: true }}>Click Me</Button>);
  expect(container.firstChild).toHaveClass("not-rounded");
});

test("applies rounded styles when no rounded prop", () => {
  const { container } = render(<Button>Click Me</Button>);
  expect(container.firstChild).toHaveClass("rounded");
});

test("applies size styles when size prop is small", () => {
  const { container } = render(<Button options={{ size: "sm" }}>Click Me</Button>);
  expect(container.firstChild).toHaveClass("px-2 py-1 text-sm");
});

test("applies size styles when size prop is small using Sizes", () => {
  const { container } = render(<Button options={{ size: Sizes.Sm }}>Click Me</Button>);
  expect(container.firstChild).toHaveClass("px-2 py-1 text-sm");
});

test("triggers onClick handler when clicked", () => {
  const handleClick = vi.fn();
  render(<Button onClick={handleClick}>Click Me</Button>);
  fireEvent.click(screen.getByText("Click Me"));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
