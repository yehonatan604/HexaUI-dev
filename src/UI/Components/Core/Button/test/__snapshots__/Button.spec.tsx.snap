// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="m-[1px] pb-[6px] py-[2.75px] rounded px-4 py-3 text-base bg-info text-white hover:bg-opacity-80 focus:outline-none focus:ring-2 focus:ring-opacity-50 focus:ring-gray-500 undefined"
  >
    Click Me
  </button>
</DocumentFragment>
`;
